azext_iot/operations/hub.py

```diff
@@ -181,6 +202,127 @@ def iot_device_delete(cmd, device_id, hub_name=None, resource_group_name=None, l
+    parent_device_id = device_scope[len(DEVICE_DEVICESCOPE_PREFIX):device_scope.rindex('-')]
+    return _iot_device_show(target, parent_device_id)
+
+
+def iot_device_set_parent(cmd, device_id, parent_id, force=False, hub_name=None, resource_group_name=None, login=None):
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name, login=login)
+    parent_device = _iot_device_show(target, parent_id)
+    _validate_edge_device(parent_device)
+    child_device = _iot_device_show(target, device_id)
+    _validate_nonedge_device(child_device)
+    _validate_parent_child_relation(child_device, parent_device['deviceScope'], force)
+    _update_nonedge_devicescope(target, child_device, parent_device['deviceScope'])
+
+
+def iot_device_children_add(cmd, device_id, child_list, force=False, hub_name=None,
+                            resource_group_name=None, login=None):
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name, login=login)
+    devices = []
+    edge_device = _iot_device_show(target, device_id)
+    _validate_edge_device(edge_device)
+    for non_edge_device_id in child_list.split(','):
+        nonedge_device = _iot_device_show(target, non_edge_device_id.strip())
+        _validate_nonedge_device(nonedge_device)
+        _validate_parent_child_relation(nonedge_device, edge_device['deviceScope'], force)
+        devices.append(nonedge_device)
+
+    for device in devices:
+        _update_nonedge_devicescope(target, device, edge_device['deviceScope'])
+
+
+def iot_device_children_remove(cmd, device_id, child_list=None, remove_all=False, hub_name=None,
+                               resource_group_name=None, login=None):
+    target = get_iot_hub_connection_string(cmd, hub_name, resource_group_name, login=login)
+    devices = []
+    if remove_all:
+        result = _iot_device_children_list(cmd, device_id, hub_name, resource_group_name, login)
+        if not result:
+            raise CLIError('No registered child devices found for "{}" edge device.'.format(device_id))
+        for non_edge_device_id in ([str(x['deviceId']) for x in result]):
+            nonedge_device = _iot_device_show(target, non_edge_device_id.strip())
+            devices.append(nonedge_device)
+    elif child_list:
+        edge_device = _iot_device_show(target, device_id)
+        _validate_edge_device(edge_device)
+        for non_edge_device_id in child_list.split(','):
+            nonedge_device = _iot_device_show(target, non_edge_device_id.strip())
+            _validate_nonedge_device(nonedge_device)
+            _validate_child_device(nonedge_device)
+            if nonedge_device['deviceScope'] != edge_device['deviceScope']:
+                raise CLIError('The entered child device "{}" isn\'t assigned as a child of edge device "{}"'
+                               .format(non_edge_device_id.strip(), device_id))
+            if nonedge_device['deviceScope'] == edge_device['deviceScope']:
```

Comment:
Do you need this `if`, seems like this check can be an `else`....then remove the `continue` couple lines down since there is no additional work being done.

Reply:
Fixed